### PR TITLE
add npm step with blackduck

### PR DESCRIPTION
**Pull request by dasormeter**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -25,9 +25,26 @@ jobs:
           path: target/daml-test-reports
       - store_artifacts:
           path: target/daml-test-reports
+          
+  npm_test:       
+    executor: daml-executor
```
Comment (bartfaitamas) [inline .circleci/config.yml]: after adding a node executor
```
executors:
  daml-executor:
    docker:
      - image: digitalasset/daml-sdk:0.12.15-circleci-master
  node-executor:
    docker:
      - image: node:8.12-alpine
```
then

```suggestion
    executor: node-executor
```

so we only need npm install below

File: .circleci/config.yml [hunk 2]
```diff
@@ -25,9 +25,26 @@ jobs:
...
+    steps:
+      - checkout
+      - run:
+          command: |
+            curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.34.0/install.sh | bash
+            nvm install v8.12.0
+            nvm use v8.12.0
+            npm install -g yarn
```
Comment (bartfaitamas) [inline .circleci/config.yml]: I don't know how exactly blackduck works, do we need to build the app first, to trigger dependency resolution? If so, then we should use a bash snippet:

```
npm install -g yarn
cd frontend
yarn run upgrade
yarn run build
```

assuming the node docker images doesn't contain the make command

Comment (bartfaitamas) [reply .circleci/config.yml]: We could use `node:8.12-stretch` docker image in the executor above, which contains make. Then we can simply

```
npm install -g yarn
make build
```

**Comment by gaborh-da** [comment]
No npm anymore.

**Pull request closed without merging**